libraries/node-core-library/src/FileSystem.ts

```diff
@@ -0,0 +1,432 @@
+import * as pathUtilities from 'path';
+import * as fs from 'fs';
+import * as fsx from 'fs-extra';
+
+import { Text } from './';
+
+/**
+ * The allowed types of encodings, as supported by Node.js
+ * @public
+ */
+export type Encoding = 'utf8';
+
+/**
+ * Available file handle opening flags.
+ * @public
+ */
+export type FileHandleOpenFlags = 'r' | 'r+' | 'rs+' | 'w' | 'wx' | 'w+' | 'wx+' | 'a' | 'ax' | 'a+' | 'ax+';
+
+/**
+ * Enumeration controlling conversion of newline characters.
+ * @public
+ */
+export enum NewlineConversion {
```

Comment:
>NewlineConversion [](start = 12, length = 17)

Rush has a scenario where when it bumps versions in package.json, it really should preserve whatever newlines Git put in that file. 

I envision the (out of scope for this PR) feature  like this:

```javascript
const previousNewlineKind: NewlineKind | undefined = undefined;
if (previousFileContents) {
  previousNewlineKind = Text.guessNewlineKind(previousFileContents);
}
// Overwrite the previous package.json
FileSystem.writeFile(filename, newContent, previousNewlineKind || NewLineKind.CrLf);
```